core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,497 @@
+  };
+
+  constructor(
+    private readonly languageUtilService: LanguageUtilService,
+    private readonly contributionAndReviewStatsService:
+        ContributionAndReviewStatsService,
+    private readonly userService: UserService) {
+  }
+
+  async ngOnInit(): Promise<void> {
+    const userInfo = await this.userService.getUserInfoAsync();
+    const username = userInfo.getUsername();
+
+    if (username === null) {
+      throw new Error('Cannot fetch username.');
+    }
+    this.username = username;
+    const currentOption = this.options.filter(
+      (option) => option.contributionType === this.type);
+    this.selectedContributionType = currentOption[0].displayName;
+
+    const userContributionRights =
+      await this.userService.getUserContributionRightsDataAsync();
+    this.userCanReviewTranslationSuggestions = (
+      userContributionRights
+        .can_review_translation_for_language_codes.length > 0);
+    this.userCanReviewQuestionSuggestions = (
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+
+    if (this.userCanReviewTranslationSuggestions) {
+      this.options.push(this.translationReviewOption);
+    }
+    if (this.userCanSuggestQuestions) {
+      this.options.push(this.questionContributionOption);
+    }
+    if (this.userCanReviewQuestionSuggestions) {
+      this.options.push(this.questionReviewOption);
+    }
+
+    await this.fetchStats();
+  }
+
+  toggleDropdown(): void {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  toggleMobileDropdown(): void {
+    this.mobileDropdownShown = !this.mobileDropdownShown;
+  }
+
+  async selectOption(contributionType: string): Promise<void> {
+    this.type = contributionType;
+    const currentOption = this.options.filter(
+      (option) => option.contributionType === contributionType);
+    this.selectedContributionType = currentOption[0].displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    this.translationContributions = [];
+    this.translationReviews = [];
+    this.questionContributions = [];
+    this.questionReviews = [];
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      const translationContributionLanguages = this.fetchLanguagesFromStats(
+        response.translation_contribution_stats);
+      translationContributionLanguages.forEach((language: string) => {
+        const translationContributionsStats:
+        TranslationContributionStats[] = [];
+        response.translation_contribution_stats.map((stat) => {
+          if (stat.language_code === language) {
+            translationContributionsStats.push(
+              this.createTranslationContributionStat(stat)
+            );
+          }
+        });
```

Comment:
Why do you need to get all the contribution languages? Can't you just iterate over all the stats objects directly? Ditto below.

Reply:
Yeah removed iterating over all the languages. Created `{[key: string]: Stats}` type objects to keep stats of a given language. We are showing stats of all languages in different containers of the same window.  I used a `Stats` to track each language's page position. If we just increment the array indexes, going into a new page in one language's statistics will result in moving pages of ever other languages' statistics. Please correct me if I am wrong.

I have updated the pagination logic to simplify it. There will be no more `PagedStats` interface. `Stats` objects will keep the starting and ending positions of the current page.